src/bundle-manifest.ts

```diff
@@ -286,19 +286,47 @@ function getBundleEntrypoint(bundle: Bundle): UrlString|null {
 }
 
 /**
- * A simple function for generating shared bundle names based on a counter.
+ * Creates a bundle url mapper function which maps non-shared bundles to the
+ * urls of their single entrypoint and yields responsibility for naming
+ * remaining shared bundle urls to the `mapper` function argument.  The
+ * mapper function takes a collection of shared bundles and a url map, calling
+ * `.set(url, bundle)` for each.
  */
-export function sharedBundleUrlMapper(bundles: Bundle[]):
-    Map<UrlString, Bundle> {
-  let counter = 0;
-  const urlMap = new Map<UrlString, Bundle>();
-  for (const bundle of bundles) {
-    const bundleEntrypoint = getBundleEntrypoint(bundle);
-    if (bundleEntrypoint) {
-      urlMap.set(bundleEntrypoint, bundle);
-    } else {
-      urlMap.set(`shared_bundle_${++counter}.html`, bundle);
+export function generateSharedBundleUrlMapper(
```

Comment:
Create a type alias for the `SharedBundleUrlMapper` function signature to be the return value of this function